### PR TITLE
Add Appointments

### PR DESCRIPTION
**Pull request by BaVilius**
- added appointments.
- this time moved the repository operations into the service layer. But in case we decide to follow the proper repository pattern I kept the same code in repository layer as well.
- skipped on doing proper validating for this PR. Will do it if we manage to pass the minimal requirements before deadline.

### CODE REVIEW

File: PspPos/Models/DTO/Requests/AppointmentCreateRequest.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿namespace PspPos.Models.DTO.Requests
```
Comment (naglissul) [inline PspPos/Models/DTO/Requests/AppointmentCreateRequest.cs:1]: i guess patiem reiks biski pamovint kai kuriuos modelius i sita folderi, nes iki siol turejom Model, ViewModel, PostModel viska prie Models

File: PspPos/Controllers/ServiceController.cs [hunk 2]
```diff
@@ -187,5 +182,151 @@ public async Task<ActionResult<ServiceDiscount>> AddDiscountToService(Guid compa
...
+        public async Task<ActionResult<Appointment>> CreateAppointment([FromBody] AppointmentCreateRequest body, Guid companyId)
+        {
+            if (await _context.CheckIfCompanyExists(companyId))
+            {
+                return StatusCode(StatusCodes.Status404NotFound, "Provided Company does not exist");
+            }
+
+            // add validation for dates
+            var appointment = new Appointment {
+                Id = Guid.NewGuid(),
```
Comment (naglissul) [inline PspPos/Controllers/ServiceController.cs:220]: idk, usually the database creates guid

**Review by naglissul — APPROVED** [review]
ig optional comments. However, haven't tested this

File: PspPos/Controllers/ServiceController.cs [hunk 3]
```diff
@@ -187,5 +182,151 @@ public async Task<ActionResult<ServiceDiscount>> AddDiscountToService(Guid compa
...
+                StartDate = DateTime.Parse(body.StartDate),
+                EndDate = DateTime.Parse(body.EndDate),
+                WorkerId = body.WorkerId,
+                StoreId = body.StoreId,
+            };
+           
+
+            await _appointmentsService.InsertAsync(appointment);
+
+            return StatusCode(StatusCodes.Status200OK);
```
Comment (TomasKvaraciejus) [inline PspPos/Controllers/ServiceController.cs:232]: O kodel cia StatusCode(...) o ne tiesiog NotFound()?

**Pull request merged**